### PR TITLE
Non painting platform views

### PR DESCRIPTION
**Pull request by hterkelsen**
If a platform view is declared as invisible, do not create an overlay for it.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart [hunk 1]
```diff
@@ -92,6 +92,8 @@ class HtmlViewEmbedder {
   /// The list of view ids that should be composited, in order.
   List<int> _compositionOrder = <int>[];
 
+  int _numVisibleViews = 0;
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart:95]: nit: `visibleViewCount` (https://dart.dev/guides/language/effective-dart/design#avoid-abbreviations)

File: lib/web_ui/lib/src/engine/platform_views/content_manager.dart [hunk 2]
```diff
@@ -186,6 +194,20 @@ class PlatformViewManager {
...
   }
 
+  /// Returns `true` if the given [viewId] is for an invisible platform view.
+  bool isInvisible(int viewId) {
+    final String? viewType = _viewIdToType[viewId];
+    if (viewType != null) {
+      return _invisibleViews.contains(viewType);
+    } else {
+      return false;
+    }
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/platform_views/content_manager.dart:204]: nit: `return viewType != null && _invisibleViews.contains(viewType);`

File: lib/web_ui/lib/ui.dart [hunk 3]
```diff
@@ -49,7 +49,9 @@ void webOnlyInitializeEngine() {
   engine.initializeEngine();
 }
 
-void webOnlySetPluginHandler(Future<void> Function(String, ByteData?, PlatformMessageResponseCallback?) handler) {
+void webOnlySetPluginHandler(
+    Future<void> Function(String, ByteData?, PlatformMessageResponseCallback?)
+        handler) {
```
Comment (yjbanov) [inline lib/web_ui/lib/ui.dart:54]: Let's revert the auto-formatted changes. The engine has switched to the main Flutter style guide. I would also recommend turning off automatic formatting in the IDE:

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#:~:text=(The%20dart%20format%20tool%20uses%20the%20Dart%20guide%2C%20so%20we%20do%20not%20use%20it%20in%20the%20flutter/flutter%20and%20flutter/engine%20repositories.%20However%2C%20we%20do%20recommend%20its%20use%20in%20general.)

**Review by yjbanov — APPROVED** [review]

File: lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart:95]: Done

File: lib/web_ui/lib/src/engine/platform_views/content_manager.dart (hunk 2 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/platform_views/content_manager.dart:204]: Done

File: lib/web_ui/lib/ui.dart (hunk 3 above)
Comment (hterkelsen) [reply lib/web_ui/lib/ui.dart:54]: done

**Pull request merged**